File: php/features.php
<?php
declare(strict_types=1);

// Pagespeed SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class PagespeedFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new PagespeedBaseFeature();
            case "test":
                return new PagespeedTestFeature();
            default:
                return new PagespeedBaseFeature();
        }
    }
}

File: py/features.py
# Pagespeed SDK feature factory

from feature.base_feature import PagespeedBaseFeature
from feature.test_feature import PagespeedTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PagespeedBaseFeature(),
        "test": lambda: PagespeedTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
